lisa/base_tools/sed.py

```diff
@@ -53,3 +53,18 @@ def append(
             shell=True,
         )
         result.assert_exit_code(message=result.stdout)
+
+    def substitute_or_append(
```

Comment:
It looks not bring much value to the method. The `is_substitute` is still a flag, which needs to be calculated outside of method. Can you include the check logic in this method? Ideally, the regexp should be used by search and substitute both.